Setup/InstallData.php

```diff
@@ -26,35 +33,46 @@ class InstallData implements InstallDataInterface
     private $rulesFactory;
 
     /**
-     * Logger
-     *
-     * @var logger
+     * Logging instance
+     * 
+     * @var \Klaviyo\Reclaim\Logger\Logger
      */
-    protected $logger;
+    protected $_klaviyoLogger;
  
     const KLAVIYO_ROLE_NAME = 'Klaviyo';
 
     /**
      * Init
      *
+     * @param DirectoryList $dir
      * @param RoleFactory $roleFactory
      * @param RulesFactory $rulesFactory
-     * @param Logger $logger
+     * @param \Klaviyo\Reclaim\Logger\Logger $klaviyoLogger
      */
     public function __construct(
+        DirectoryList $dir,
         RoleFactory $roleFactory,
         RulesFactory $rulesFactory,
-        \Psr\Log\LoggerInterface $logger
+        \Klaviyo\Reclaim\Logger\Logger $klaviyoLogger
     ) {
+        $this->_dir = $dir;
         $this->roleFactory = $roleFactory;
         $this->rulesFactory = $rulesFactory;
-        $this->logger = $logger;
+        $this->_klaviyoLogger = $klaviyoLogger;
     }
  
     public function install(
         ModuleDataSetupInterface $setup, 
         ModuleContextInterface $context
     ) {
+        //Klaviyo log file creation
+        $path = $this->_dir->getPath('log') . '/klaviyo.log';
```

Comment:
i feel like this is kinda reused all of the place, the getting of the path and such, is this able to be isolated?